### ISSUE TITLE
Migrate to travis

### ISSUE THREAD

**Issue by scopatz** [cyclus/cyclus#1117]
Steps:

1. Turn off moth site
2. copy `.travis.yml` file to root dir
3. modify travis file to build from the ciclus repo, rather than pyne-ci
4. modify travis file to run cyclus tests in the script section
5. modify the ciclus meta.yml files to use the travis selectors, similar to those in pyne-ci
6. open to see if it works
7. remove the site from moth.

read the travis and conda docs for more info.

**Comment by gidden** [comment]
Current PRs testing with an updated ciclus:
- https://github.com/gidden/cyclus/pull/16
- https://github.com/gidden/cycamore/pull/7

**Comment by gidden** [comment]
@scopatz, are travis selectors incompatible with polyphemus? See [this](http://submit-1.batlab.org/nmi-runs/cyclusci/2015/04/cyclusci_submit-1.batlab.org_1429624072_6164/userdir/nmi:x86_64_MacOSX8/remote_pre_declare.err) failure for #1127.

**Comment by scopatz** [comment]
No, they shouldn't be.  On BaTLab that expression evaluates to `False`. I suspect that there is something else that is broken.  For instance, why is the install trying to find 0.2 rather than the latest version (0.0):

```console
+ WORKDIR=anaconda/conda-bld/work/gidden-cyclus-b2e244c
++ cat cyclus/meta.yaml
++ grep version
+ vers='  version: "0.2"'
+ read -a versArray
+ anaconda/bin/conda install --use-local 'cyclus="0.2"'
Error: No packages found in current linux-64 channels matching: cyclus "0.2"*

You can search for this package on Binstar with

    binstar search -t conda cyclus "0.2"*

```
http://submit-1.batlab.org/nmi-runs/cyclusci/2015/04/cyclusci_submit-1.batlab.org_1429624072_6164/userdir/nmi:x86_64_Ubuntu12/remote_pre_declare.err

**Comment by gidden** [comment]
The mac run on 1127 is still failing this morning with the same error:
```
anaconda/bin/conda build --no-test cyclus
Error: Invalid selector in meta.yaml line 7:
  fn: cyclus-src.tar.gz # ["TRAVIS" not in environ]
```

**Comment by gidden** [comment]
(http://submit-1.batlab.org/nmi/results/details?runID=315884)

**Comment by gidden** [comment]
On the travis front, it looks like we're close. There's currently a linking error with hdf5 during test execution: 
```
/home/travis/miniconda/bin/cyclus_unit_tests_base: error while loading shared libraries: libhdf5.so.8: cannot open shared object file: No such file or directory
```

See [here](https://travis-ci.org/gidden/cyclus/jobs/59552497) for instance.

This looks to be a conda/hdf problem. I set the `LD_LIBRARY_PATH` to the conda lib directory, but that does not fix it. Any ideas?

**Comment by scopatz** [comment]
1. Why are we worrying about mac CI? I thought we were abandoning it.
2. This is the linking problem that has been discussed on the list (https://groups.google.com/d/topic/cyclus-users/G-9xbOgPeU4/discussion).  I believe that this is caused by our CMake building finding the system libraries if they are installed, rather than the conda ones.  This could be fixed in the conda recipe but the correct fix is to update the build system.

**Comment by gidden** [comment]
The build system is finding the correct [hdf5](https://travis-ci.org/gidden/cyclus/jobs/59579041#L378). The problem is that the install prefix during the conda build is `miniconda/envs/build_` not `miniconda`.

**Comment by scopatz** [comment]
> The problem is that the install prefix during the conda build is miniconda/envs/build_ not miniconda. 

That is how conda works.  It then adjusts the paths in the libraries on install.

**Comment by scopatz** [comment]
Try throwing in an `ldd $(which cyclus_unit_tests_base)` after install to see what is being linked.

**Comment by gidden** [comment]
Note, there is hdf5 on the system ([6.0.3](https://travis-ci.org/gidden/cyclus/jobs/59607662#L2023)), but we are not picking it up

**Comment by gidden** [comment]
I also used the `HDF5_ROOT` CMake variable. The include directory is [correct](https://travis-ci.org/gidden/cyclus/jobs/59607662#L379).

**Comment by gidden** [comment]
closed by #1130
